c7n/resources/redshift.py

```diff
@@ -660,6 +662,35 @@ class RedshiftSnapshotAge(AgeFilter):
     date_attribute = 'SnapshotCreateTime'
 
 
+@RedshiftSnapshot.filter_registry.register('cross-account')
+class RedshiftSnapshotCrossAccount(CrossAccountAccessFilter):
+    """Filter all accounts that allow access to non-whitelisted accounts
+    """
+    permissions = ('redshift:DescribeClusterSnapshots',)
+    schema = type_schema(
+        'cross-account',
+        whitelist={'type': 'array', 'items': {'type': 'string'}},
+        whitelist_from=ValuesFrom.schema)
+
+    def process(self, snapshots, event=None):
+        self.accounts = self.get_accounts()
+        orig_len = len(snapshots)
+        snapshots = [s for s in snapshots if s.get('AccountsWithRestoreAccess')]
+        self.log.info(
+            'Implicitly filtered out %s of %s snapshots with no '
```

Comment:
fwiw, a filter doesn't implicitly filter, a filter is always explicit, we do these warning/info notices within actions which are not supposed to be filtering, since we're generating resource outputs and metrics based on whats passed through a policy's filters.